### PR TITLE
Fix zlib compilation on conan and AppleClang 12.0

### PR DESCRIPTION
**Pull request by yamadapc**
I'm not sure what these replaces were doing.

Happy to create a new recipe if that is a safer change, since I just don't know what the replaces were doing, this just fixed it on my computer™️.

c.c. @lasote who wrote the recipe.

Related to:

- https://github.com/boostorg/build/issues/638
- https://github.com/madler/zlib/pull/509

Introduced in https://github.com/conan-io/conan-center-index/pull/1

Specify library name and version:  **zlib/1.2.11**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/zlib/1.2.11/conanfile.py [hunk 1]
```diff
@@ -89,15 +89,6 @@ def _build_zlib(self):
             tools.replace_in_file('gzguts.h',
                                   '#if defined(_WIN32) || defined(__CYGWIN__)',
                                   '#if defined(_WIN32) || defined(__MINGW32__)')
```
Comment (uilianries) [inline recipes/zlib/1.2.11/conanfile.py:91]:
```suggestion
        if self.settings.compiler != "apple-clang" or tools.Version(self.settings.compiler.version) < "11":
```

Do not remove it, instead, add condition to skip apple-clang 12

Comment (uilianries) [inline recipes/zlib/1.2.11/conanfile.py:91]:
```suggestion
        if self.settings.compiler != "apple-clang" or tools.Version(self.settings.compiler.version) < "11":
```

Do not remove it, instead, add condition to skip apple-clang 12

**Comment by uilianries** [comment]
@yamadapc Please, ask for EAP inclusion in #4, otherwise the CI bot won't run this PR.

File: recipes/zlib/1.2.11/conanfile.py (hunk 1 above)
Comment (yamadapc) [reply recipes/zlib/1.2.11/conanfile.py:91]: I suppose it should be `< "12"`? not `< "11"`.

Will make the patch.

File: recipes/zlib/1.2.11/conanfile.py [hunk 2]
```diff
@@ -89,15 +89,18 @@ def _build_zlib(self):
...
-                tools.replace_in_file(filename,
-                                      '#ifdef HAVE_UNISTD_H    '
-                                      '/* may be set to #if 1 by ./configure */',
-                                      '#if defined(HAVE_UNISTD_H) && (1-HAVE_UNISTD_H-1 != 0)')
-                tools.replace_in_file(filename,
-                                      '#ifdef HAVE_STDARG_H    '
-                                      '/* may be set to #if 1 by ./configure */',
-                                      '#if defined(HAVE_STDARG_H) && (1-HAVE_STDARG_H-1 != 0)')
+
+            is_apple_clang12 = self.settings.compiler == "apple-clang" and tools.Version(self.settings.compiler.version) >= "12.0"
```
Comment (yamadapc) [inline recipes/zlib/1.2.11/conanfile.py:93]: It should now run the replaces unless on `AppleClang >= "12.0"`

**Comment by yamadapc** [comment]
Forced push to re-trigger the build. Please have another look.

Thanks

**Pull request merged**